napari/_tests/utils.py

```diff
@@ -120,9 +121,43 @@ def add_layer_by_type(viewer, layer_type, data, visible=True):
     return layer2addmethod[layer_type](viewer, data, visible=visible)
 
 
+def are_objects_equal(object1, object2):
+    """
+    compare two (collections of) arrays or other objects for equality. Ignores nan.
+    """
+    if isinstance(object1, (list, tuple)):
+        items = zip(object1, object2)
+    elif isinstance(object1, dict):
+        items = [(value, object2[key]) for key, value in object1.items()]
+    else:
+        items = [(object1, object2)]
+
+    # equal_nan does not exist in array_equal in old numpy
+    if tuple(int(v) for v in np.__version__.split('.')) < (1, 19):
+        fixed = []
+        for a1, a2 in items:
+            a1 = np.asarray(a1)
+            a2 = np.asarray(a2)
+            if a1.dtype.kind == 'f':
+                a1 = a1[np.isfinite(a1)]
+            if a2.dtype.kind == 'f':
+                a2 = a2[np.isfinite(a2)]
+            fixed.append((a1, a2))
+        return np.all([a1 == a2 for a1, a2 in items])
```

Comment:
Maybe `np.nan_to_num` is a better fix here? I just checked and it passes through non-float dtypes unchanged. So this would be:

```python
fixed = [(np.nan_to_num(a1), np.nan_to_num(a2)) for a1, a2 in items]
```

Also, it would be good if you actually used `fixed`, rather than ignoring it and using `items`... 😂

Reply:
hah, that was the problem! :joy: